src/org/sosy_lab/java_smt/example/ListAllSolversWithFeatures.java

```
@@ -0,0 +1,311 @@
+    try (InterpolatingProverEnvironment<?> prover =
+        context.newProverEnvironmentWithInterpolation(ProverOptions.GENERATE_MODELS)) {
+      if (prover != null) {
+        features.add("Interpolation");
+      }
+    } catch (UnsupportedOperationException e) {
+    }
+
+    // UnsatCore: throws UnsupportedOperationException if not available.
+    // TODO: check behavior with null argument properly
+    try (ProverEnvironment prover =
+        context.newProverEnvironment(ProverOptions.GENERATE_UNSAT_CORE)) {
+      if (prover.getUnsatCore() != null) {
+        features.add("UnsatCore");
+      }
+    } catch (UnsupportedOperationException e) {
+    } catch (Exception e) {
+      features.add("UnsatCore");
+    }
+
+    // UnsatCoreOverAssumptions: throws NullPointerException if available.
+    try (ProverEnvironment prover =
+        context.newProverEnvironment(GENERATE_UNSAT_CORE_OVER_ASSUMPTIONS)) {
+      prover.unsatCoreOverAssumptions(null);
+    } catch (UnsupportedOperationException e) {
+    } catch (NullPointerException e) {
+      features.add("UnsatCore /w Assumption");
+    }
+
+    // TODO: Remove or rework to check if solver overrides!
+    // AllSat: All solvers support this through JavaSMT.
+    try (ProverEnvironment prover = context.newProverEnvironment(ProverOptions.GENERATE_ALL_SAT)) {
+      try {
+        if (prover.allSat(
+                new AllSatCallback<>() {
+
+                  List<List<BooleanFormula>> models = new ArrayList<>();
+
+                  @Override
+                  public void apply(List<BooleanFormula> pModel) {
+                    models.add(pModel);
+                  }
+
+                  @Override
+                  public List<List<BooleanFormula>> getResult() {
+                    return models;
+                  }
+                },
+                ImmutableList.of())
+            != null) {
+          features.add("AllSAT");
+        }
+      } catch (UnsupportedOperationException e) {
+      }
```

add a comment, why the exception can be ignored.